File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.cpp
class Solution {
public:
    string reverseWords(string s) {
        
        
        int j=0;
        
    string ans;
        
        for(int i=0;i<s.length();)
        {
            while(s[i]==' ' && i<s.length())
            {
                i++;
            }
            
            if(i>=s.length())break;
            
            j=i;
            
            while(s[j]!=' ' && j<s.length())
            {
                j++;
            }
            
            
            string p=s.substr(i,j-i);
          
            if(ans.length()==0) ans=p;
            
            else
                ans=p+' '+ans;
            
            i=j+1;
                
        }
        return ans;
    }
};

File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.py
class Solution:
   
 def reverseWords(self, s: str) -> str:
    ans = ""
    i = 0
    n = len(s)
    
    while i < n:
        while i < n and s[i] == ' ':
            i += 1
        
        if i >= n:
            break
                
        j = i
        while j < n and s[j] != ' ':
            j += 1
            
        p = s[i:j]
        
        if ans=="":
            ans = p
        else:
            ans = p + ' ' + ans
            
        i = j + 1
        
    return ans